### ISSUE TITLE
Using cy.clock(), Graphs does not gets plotted during the test case execution.

### ISSUE THREAD

**Issue by Ppkd2021** [cypress-io/cypress#17190]
### Current behavior
There is a scenario where we need to validate the data plotted on the graph for last 30 days of data. The data keeps on changing after every 30 days.
We mock the response for the 30 days of data by setting clock before we visit that page to be in the response timeframe. 
What we observed is if we mock response by setting cy.clock then graph does not get plotted and if we simply mock the latest data without setting clock then graphs get plotted.
Without Cy.Clock()
![image](https://user-images.githubusercontent.com/59794856/124380559-e3eeee80-dcda-11eb-889e-b0400b77f724.png)
With Cy.Clock()
![image](https://user-images.githubusercontent.com/59794856/124380584-11d43300-dcdb-11eb-88c0-e06f5bcb9bde.png)

### Desired behavior
We tried debugging from the application perspective while test cases are running and we found out that the component that displays the data is receiving the data successfully but not able to know the reason why on cypress iframe browser it is not getting plotted.
We are using high charts library to plot the data in the application.

### Test code to reproduce
```
const path = `mock-responses/group-conversation.spec`;
		cy.MockQueries(`ListGroupKeywordMetricsByGroupId`, path);
		cy.MultipleInterceptRoute([`GetLastdayGroupMetricsByGroupId`, `FetchConversations`, `ListGroupMetricsByGroupId`]);
		cy.restoreLocalStorage();
		cy.visit(`group-admin/group/be52327e-2310-4cd0-8b87-2231e8b3def9/conversationtrends#7days`);
		cy.clock(Date.UTC(2021, 6,4));
```

### Versions

Cypress version 7.6.0

**Comment by Ppkd2021** [comment]
The above behaviour is happening throughout the application, As we have around 10-12 graphs that we tried validating via cypress.

**Comment by jennifer-shehane** [comment]
`cy.clock()` overrides many timer funtions. So your graphs may be dependent on some of the other timer functions not being stubbed. https://on.cypress.io/clock

I would try specifying the exact time function you want to override for your test case so that it doesn't override all the others. Example here: https://docs.cypress.io/clock#Specify-which-functions-to-override

**Comment by Ppkd2021** [comment]
@jennifer-shehane Thanks for the information you shared that helped us to find the solution for the above issue.

**Comment by jennifer-shehane** [comment]
Great to hear! Closing as resolved. 

If you're experiencing a bug similar to this in Cypress, please [open a new issue](https://github.com/cypress-io/cypress/issues/new/choose) with a fully reproducible example that we can run. There may be a specific edge case with the issue that we need more detail to fix.

**Comment by Ppkd2021** [comment]
@jennifer-shehane Actually, We tried another graph and it does not solve the problem there. 
Case 1 - When we are using cy.clock with Date as an overriding function then the graph dates are coming right but data is not getting plotted. The date which is highlighted is coming correct as we have set up in cy.clock.
![image](https://user-images.githubusercontent.com/59794856/125163754-46009580-e1ac-11eb-90a4-a06cd47f4e11.png)
Case 2 - overriding without date but other functions then data is coming under graph but the date is not coming as set in cy.clock.
